app/js/config.js

```diff
@@ -9,6 +9,10 @@ app.config(function ($stateProvider, $locationProvider, $urlRouterProvider, $mdT
         .primaryPalette('indigo')
         .accentPalette('orange');
 
+    const carregaDisciplina = function(Disciplina, id) {
+        return new Disciplina(id);
```

Comment:
Não é legal utilizar new para criar um objeto. Creio que deve ser algo a ser refatorado.

Reply:
Por que não é legal? Uma das maneiras de criar um objeto em JS é criando uma função que retorna um objeto. Disciplina é uma factory que retorna uma função de criar um objeto, por isso o new.

Reply:
Pq para criação de objeto em JS não se precisa de 'new'. Implicitamente qnd vc chema um new ele cria um objeto vazio aponto para o this. Isso é fonte de muito bugs, pq o diz não se refere apenas ao escopo da função que ele tá inclusa (como acontece com linguagens OO). Em java script as funções são de primeira ordem (tudo função) e não há necessidade de criar um construtor. O 'jeito' correto (q não há jeito correto) é utilizar Object.create, Object. assign. Alguns links pra tu vê mais: http://raganwald.com/2014/01/14/the-new-javascript-problem.html, https://www.liip.ch/en/blog/why-i-dont-use-the-javascript-new-keyword

Reply:
Mas assim, há algumas situações em que n da pra fugir do new, como a criação de formDatas, entre outras coisas.

Reply:
Mas isso se aplica a criação de services não singleton em angular? O this que tiver fora do prototype refere-se a "classe" mesmo e o que tiver dentro, a instância do objeto. Tu conhece algum padrão pra criação de factory sem usar o New? Ou esse create serve?

Reply:
Sim amigo, vou deixar aqui dois exemplos pra tu vê. 

Aqui é o padrão da forma mais simples (existe dezenas de padrão de criação em js): https://ideone.com/mmms8W 

Esse aqui é o padrão de módulo: https://ideone.com/1fCeu2 

A vantagem desse último é que você escolhe quais funções irão ficar expostas.